src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -1300,6 +1311,7 @@ public void writeToNBT(final NBTTagCompound data) {
         data.setTag("tasks", list);
 
         data.setTag("waitingFor", this.writeList(this.waitingFor));
+        data.setTag("waitingForMissing", this.writeList(this.waitingForMissing));
```

Comment:
Why do you need a new waitingForMissing here, is waitingFor not enough information?

Reply:
More than enough. waitingFor will include all items which currently crafting. I want to separate missing mode quene from reqular quene bacause two reasons. Performance and missing mode might extract wrong item from storage, for example wrong circuit due to weird circuit oredict and craft will stuck.

Reply:
Ah I see, I misunderstood what this nbt data actually stored, thanks for clearing it up